### PR TITLE
Update ClinVar to map YPAR to chrX and chrY

### PR DESCRIPTION
**Pull request by dglemos**
Check this ticket for more details: [ENSVAR-5483](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-5483)

### CODE REVIEW

**Review by nuno-agostinho — APPROVED** [review]
Hi @dglemos, thanks for the PR!

Just a question: the script will not add variant_features for variants from dbSNP. So if a dbSNP variant is only inserted in chrX, it won't be inserted in chrY by this script. Is that okay?

I also have some other questions/suggestions below.

**Comment by dglemos** [comment]
Thanks for the review, @nuno-agostinho 

> Just a question: the script will not add variant_features for variants from dbSNP. So if a dbSNP variant is only inserted in chrX, it won't be inserted in chrY by this script. Is that okay?

Yes, that's ok.

File: scripts/import/import_clinvar_xml [hunk 1]
```diff
@@ -471,21 +471,36 @@ sub get_feature{
...
+
     foreach my $loc(@{$seqLocs}){
 
       next unless ref($loc) eq  'HASH'; 
       next unless $loc->{Assembly} eq $assembly;
 
-      $feature{Chr}    = $loc->{Chr} ; 
-      $feature{start}  = $loc->{start} ; 
-      $feature{end}    = $loc->{stop} ; 
+      if($feature_found) {
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/import_clinvar_xml:482]: As this is clinical it's more relevant to import for chromosomes.

File: scripts/import/import_clinvar_xml [hunk 2]
```diff
@@ -471,21 +471,36 @@ sub get_feature{
...
-      $feature{measureXrefs}{$measure->{ID}}{end}   = $loc->{stop}   if $type eq $haplotype_type;
+      else{
+        $feature{Chr}       = $loc->{Chr};
+        $feature{start}     = $loc->{start};
+        $feature{end}       = $loc->{stop};
+        $feature{accession} = $loc->{Accession};
+
+        $feature{measureXrefs}{$measure->{ID}}{Chr}   = $loc->{Chr}    if $type eq $haplotype_type;
+        $feature{measureXrefs}{$measure->{ID}}{start} = $loc->{start}  if $type eq $haplotype_type;
+        $feature{measureXrefs}{$measure->{ID}}{end}   = $loc->{stop}   if $type eq $haplotype_type;
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/import_clinvar_xml:497]: The `measureXrefs` is going to be used to check if variant is in multiple locations. I think checking for the chrX is enough.

File: scripts/import/import_clinvar_xml [hunk 3]
```diff
@@ -1379,73 +1355,178 @@ sub enter_var{
     );
 
   $var_feat_adaptor->store($vf);
+  push @vfs, $vf;
+
+  # If variant in Y PAR region store the two variation features
+  #  One of the vf has already been checked
+  #  The other vf can have different chr, start, end and/or alleles (from hgvs genomic)
+  if($data->{feature_info}->{ypar} && $data->{feature_info}->{ypar}->{hgvs_g} ne "unknown:") {
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/import_clinvar_xml:1363]: In ClinVar we only look for `genomic, top level`

File: scripts/import/import_clinvar_xml [hunk 4]
```diff
@@ -1379,73 +1355,178 @@ sub enter_var{
...
+  #  The other vf can have different chr, start, end and/or alleles (from hgvs genomic)
+  if($data->{feature_info}->{ypar} && $data->{feature_info}->{ypar}->{hgvs_g} ne "unknown:") { 
+    $slice = $slice_adaptor->fetch_by_region( 'chromosome', $data->{feature_info}->{ypar}->{Chr} );
+
+    my ($ref_allele_ypar, $alt_allele_ypar) = get_hgvs_alleles( $data->{feature_info}->{ypar}->{Chr}.":".$data->{feature_info}->{ypar}->{hgvs_g} );
+    $allele_str = $ref_allele_ypar ."/". $alt_allele_ypar;
+
+    ($allele_str, $vf_start, $vf_end) = format_coords($ref_allele_ypar, $alt_allele_ypar, $allele_str, $vf_start, $vf_end, $data->{feature_info}->{ypar});
+
+    $so_term = SO_variation_class($allele_str);
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/import_clinvar_xml:1371]: In the variation table we only store one entry with one SO term. We can't save the same variant ID with different SO.

File: scripts/import/import_clinvar_xml [hunk 5]
```diff
@@ -1379,73 +1355,178 @@ sub enter_var{
     );
 
   $var_feat_adaptor->store($vf);
+  push @vfs, $vf;
+
+  # If variant in Y PAR region store the two variation features
+  #  One of the vf has already been checked
+  #  The other vf can have different chr, start, end and/or alleles (from hgvs genomic)
```
[the remark this answers is not in the document]
Comment (dglemos) [reply scripts/import/import_clinvar_xml:1363]: Yes, I have to update the script to support the new file format. As I have to refactor a few things, I can do this update too in the same PR.

**Review by nuno-agostinho — APPROVED** [review]

**Pull request merged**